src/telemetry_subscriber.rs

```diff
@@ -130,50 +153,43 @@ impl TelemetrySubscriber {
 }
 
 impl Subscriber for TelemetrySubscriber {
-    fn enabled(&self, metadata: &Metadata<'_>) -> bool {
-        metadata.level() <= &self.log_level
+    fn enabled(&self, _metadata: &Metadata<'_>) -> bool {
+        true
     }
 
     fn new_span(&self, span: &Attributes<'_>) -> Id {
         let new_span = self.build_span(span);
-        let id = loop {
-            // collision is extremely unlikely (random u64), handle anyway
-            // v. unlikely but possible: identical id is gen'd between 'contains_key' and 'insert'
-            let id = generate_span_id();
-            if !self.spans.contains_key(&id) {
-                break id;
-            };
-        };
 
-        self.spans.insert(
-            id.clone(),
-            RefCt {
+        let idx: usize = self
+            .spans
+            .insert(RwLock::new(RefCt {
```

Comment:
you might want to change the reference counting to use an atomic integer for the ref count operations, if you did this, you could then move it outside of the RwLock, so that the reference count operations can happen atomically. 

it would be fine to make this change later, it's entirely an internal optimization.